piff/gp_interp.py

```diff
@@ -320,15 +320,9 @@ def _finish_read(self, fits, extname):
         self.optimizer = data['OPTIMIZER'][0]
 
         if len(self.kernel_template)==1:
-            self.kernels = [copy.deepcopy(self.kernel_template[0]) for i in range(self.nparams)]
+            self.kernels = [self.kernel_template[0] for i in range(self.nparams)]
         else:
-            if len(self.kernel_template)!= self.nparams:
-                raise ValueError(
-                    "numbers of kernel provided should be 1 (same for all parameters) or " +
-                    "equal to the number of params (%i), number kernel provided: %i"%(
-                        (self.nparams,len(self.kernel_template))))
-            else:
-                self.kernels = [copy.deepcopy(ker) for ker in self.kernel_template]
+            self.kernels = [ker for ker in self.kernel_template]
```

Comment:
Since this is the read command, there's really no way for the number of kernels not to be either 1 or nparams at this point, so it's not worth checking and raising an exception.  If the PSF model has been run, the initialize will have already checked that things are valid before writing the output file.  Then when reading it in, it has to still be valid.